src/libraries/PoolStateLibrary.sol

```diff
@@ -358,6 +376,46 @@ library PoolStateLibrary {
         }
     }
 
+    /// @notice returns the reserves of a currency
+    /// @param manager The pool manager contract.
+    /// @param currency The currency to get the reserves for.
+    /// @return value The reserves of the currency.
+    /// @dev returns 0 if the reserves are not synced
+    /// @dev returns type(uint256).max if the reserves are synced but the value is 0
+    function getReserves(IPoolManager manager, Currency currency) internal view returns (uint256) {
+        uint256 slot = RESERVES_OF_SLOT;
+        bytes32 key;
+        assembly {
+            mstore(0, slot)
+            mstore(32, currency)
+            key := keccak256(0, 64)
+        }
+        return uint256(manager.exttload(key));
+    }
+
+    /// @notice Returns the number of nonzero deltas open on the PoolManager that must be zerod out before the contract is locked
+    function getNonzeroDeltaCount(IPoolManager manager) internal view returns (uint256) {
+        return uint256(manager.exttload(bytes32(NONZERO_DELTA_COUNT_SLOT)));
+    }
+
+    /// @notice Get the current delta for a caller in the given currency
+    /// @param caller_ The address of the caller
+    /// @param currency The currency for which to lookup the delta
+    function currencyDelta(IPoolManager manager, address caller_, Currency currency) external view returns (int256) {
+        bytes32 key;
+        assembly {
+            mstore(0, caller_)
+            mstore(32, currency)
+            key := keccak256(0, 64)
+        }
+        return int256(uint256(manager.exttload(key)));
+    }
+
+    /// @notice Returns whether the contract is unlocked or not
+    function isUnlocked(IPoolManager manager) external view returns (bool) {
+        return manager.exttload(bytes32(IS_UNLOCKED_SLOT)) != 0x0;
```

Comment:
do you think it would make sense to either have our global vars as `bytes32`, or have `exttload` accept a uint256? feels wrong to cast so often

Reply:
I'd prefer having global vars bytes32, I think bytes32 for arguments and return data for `exttload` is the most neutral option

Reply:
like lets change IS_UNLOCKED_SLOT and any other slot variables to `bytes32`